projects/ng-table-virtual-scroll/src/lib/table-data-source.ts

```diff
@@ -78,19 +82,21 @@ export class CdkTableVirtualScrollDataSource<T> extends DataSource<T> implements
           let cachedPages = new Set<number>();
           let maxPage = Number.MAX_VALUE;
           return collectionViewer.viewChange.pipe(
-            switchMap(listRange => {
-              if (listRange.start === 0 && listRange.end === Number.MAX_VALUE) return of();
-              const startPage = Math.floor(listRange.start / pageSettings.pageSize);
-              const endPage = Math.ceil(listRange.end / pageSettings.pageSize) - 1;
-
+            filter(listRange => !(listRange.start === 0 && listRange.end === Number.MAX_VALUE)),
+            map(listRange => ({
+              startPage: Math.floor(listRange.start / pageSettings.pageSize),
+              endPage: Math.ceil(listRange.end / pageSettings.pageSize) - 1,
+            })),
+            distinctUntilChanged((x, y) => x.startPage === y.startPage && x.endPage === y.endPage),
+            switchMap(paging => {
               const pages$ = generate(<GenerateBaseOptions<number>>{
-                initialState: startPage,
-                condition: x => x <= endPage && x <= maxPage,
+                initialState: paging.startPage,
+                condition: x => x <= paging.endPage && x <= maxPage,
                 iterate: x => x + 1,
               });
 
               return pages$.pipe(
-                switchMap(page => {
+                concatMap(page => {
```

Comment:
Je crois qu'on pourrait utiliser un mergemap sans issue. Ça serait potentiellement mieux d'un point de vue rapidité de résultat en plus
```suggestion
                mergeMap(page => {
```

Reply:
Effectivement! Le mergeMap semble bien fonctionner et semble encore plus quick. Thanks pour l'input!